### PR TITLE
Event page

### LINKED ISSUE #110: Events Page: Create Layout
Create the layout for the events page. See [the design doc](https://www.figma.com/file/mbJusS7ADxfO7s8lkNLc9B/GCColab-HIFI?node-id=1501%3A4952&mode=dev). Refer to the home or event page for the page layout. Use the "page-container" class to apply the correct margins. Use the app-banner component (refer to home.component.html)

This work relies on #106.

image

### PR DESCRIPTION
**Pull request by AdiMakkar**
closes #110

### CODE REVIEW

File: src/app/features/events/events.component.ts [hunk 1]
```diff
@@ -1,9 +1,38 @@
...
   templateUrl: './events.component.html',
   styleUrls: ['./events.component.scss']
 })
 export class EventsComponent {
 
+  routes = CoreRoutes;
+  @Input() form: FormGroup = new FormGroup({});
+  @Input() control!: FormControl;
+  @Input() loading: boolean = false;
+  @Input() editing: boolean = false;
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:18]: I don't think this is used in this component so you can delete it.

File: src/app/features/events/events.component.ts [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
+  events: Event[] = [];
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:19]: Not used so you can remove it.

File: src/app/features/events/events.component.ts [hunk 3]
```diff
@@ -1,9 +1,38 @@
...
+  eventsPage: number = 1;
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:20]: Not used you can remove it.

File: src/app/features/events/events.component.ts [hunk 4]
```diff
@@ -1,9 +1,38 @@
...
+  loadingEvents: boolean = true;
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:21]: Not used you can remove it.

File: src/app/features/events/events.component.ts [hunk 5]
```diff
@@ -1,9 +1,38 @@
...
+  materialButtonType = MaterialButtonType;
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:22]: Also not used so you can remove it.

File: src/app/features/events/events.component.ts [hunk 6]
```diff
@@ -1,9 +1,38 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { CoreRoutes } from 'src/app/core/constants/routes.constants';
+import { Translations } from 'src/app/core/services/translations.service';
+import { MaterialButtonType } from 'src/app/shared/models/material-button-type';
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:5]: Remove this since we aren't using it.

File: src/app/features/events/events.component.ts [hunk 7]
```diff
@@ -1,9 +1,38 @@
...
+  materialButtonType = MaterialButtonType;
+
+  constructor(public translations: Translations, 
+              public eventService: EventService) {}
+
+  confirmEvent() {
+    throw new Error('Method not implemented.');
+    }
+
+  declineEvent() {
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:31]: This method is never used so you can remove it.

File: src/app/features/events/events.component.ts [hunk 8]
```diff
@@ -1,9 +1,38 @@
...
+  @Input() editing: boolean = false;
+  events: Event[] = [];
+  eventsPage: number = 1;
+  loadingEvents: boolean = true;
+  materialButtonType = MaterialButtonType;
+
+  constructor(public translations: Translations, 
+              public eventService: EventService) {}
+
+  confirmEvent() {
```
Comment (doug0102) [inline src/app/features/events/events.component.ts:27]: This method is never used so you can remove it.

**Comment by doug0102** [comment]
Also check that the test is passing. I suspect that it will fail right now since I didn't see any changes in the file. You will need to add the stuff for the translations to work as well as that FormControl pipe. You can see an example of that in `event-form.component.spec.ts`

**Review by doug0102 — CHANGES_REQUESTED** [review]
See my comment about the test file. I think you will need to double check that the test is passing. An example of what you will probably need to add can be found in `event-form.component.spec.ts`

**Comment by AdiMakkar** [comment]
Requested changes made. Tests updated for Translations and FormControl.

**Review by doug0102 — DISMISSED** [review]

File: src/app/features/events/events.component.html [hunk 9]
```diff
@@ -1 +1,89 @@
...
+</div>
+ 
+<div class="upcoming-events-content">
+    <section>
+        <div class="upcoming-events-top">
+            <app-list 
+                [loadTime]="0"
+                [service]="eventService" 
+                [cardSize]="'large'" 
+                orientation="grid"
```
[the remark this answers is not in the document]
Comment (AdiMakkar) [reply src/app/features/events/events.component.html:28]: Orientation switched to horizontal.

**Review by doug0102 — APPROVED** [review]

**Pull request merged**